### PR TITLE
fix: docs: ieee802154: fixes issues left over from #62503

### PR DESCRIPTION
**Pull request by ghost**
@kartben @jukkar  This fixes the missing definitions in docs that started with ...section.

Also fixes some minor issues that slipped through in the recent major documentation update. The commit messages specify the details.

~~Once the build finishes, I'll check whether the problems have actually been resolved.~~

Looks better IMO but I cannot get the group ordering problem fixed (as discussed on Discord). Any ideas? If not I think the current ordering would be acceptable anyway.

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]
:+1: LGTM

File: include/zephyr/net/ieee802154.h [hunk 1]
```diff
@@ -134,23 +149,42 @@ extern "C" {
...
+ * device is not associated, see section 8.4.3, table 8-94.
+ */
 #define IEEE802154_PAN_ID_NOT_ASSOCIATED	IEEE802154_BROADCAST_PAN_ID
 
-/** interface-level security attributes, see section 9.5. */
+/** Interface-level security attributes, see section 9.5. */
 struct ieee802154_security_ctx {
-	/** section 9.5, secFrameCounter */
+	/**
+	 * The interface-level outgoing frame counter, section 9.5, table 9-8,
```
Comment (kartben) [inline include/zephyr/net/ieee802154.h:182]: drop The
```suggestion
	 * Interface-level outgoing frame counter, section 9.5, table 9-8,
```

File: include/zephyr/net/ieee802154.h [hunk 2]
```diff
@@ -97,8 +97,23 @@ extern "C" {
...
+ * @note Currently we only support 127 byte sized packets although some PHYs
+ * (e.g. SUN, MSK, LECIM, ...) support larger packet sizes. Needs to be changed
+ * once we want to fully support those PHYs.
+ */
+#define IEEE802154_MAX_PHY_PACKET_SIZE	127
+
+/**
+ * @brief Represents the frame check sequence length, see section 7.2.1.1.
+ *
+ * @note Currently we only support a 2 byte FCS although some PHYs (e.g. SUN,
```
Comment (kartben) [inline include/zephyr/net/ieee802154.h:112]: I am not sure about the use of "we" (here and in a few other places in this patch, plus in other recent changes too it seems). In other places you have something along the lines of "Currently only XXX is [partially] supported" and I feel it's more appropriate.
I wouldn't let this prevent this PR to go through though.

File: include/zephyr/net/ieee802154.h [hunk 3]
```diff
@@ -171,24 +205,23 @@ struct ieee802154_security_ctx {
 	 */
 	uint8_t key[16];
 
-	/** frame-level security key material */
+	/** Length in byptes of the interface-level security key material. */
```
Comment (kartben) [inline include/zephyr/net/ieee802154.h:208]:
```suggestion
	/** Length in bytes of the interface-level security key material. */
```

File: include/zephyr/net/ieee802154_radio.h [hunk 4]
```diff
@@ -620,194 +621,208 @@ enum ieee802154_fpb_mode {
...
+	 *    short and extended addresses of the peer to which it should inject CSL IEs.
+	 * 2. Use @ref IEEE802154_CONFIG_CSL_RX_TIME periodically, before each use of
+	 *    @ref IEEE802154_CONFIG_CSL_PERIOD setting parameters of the nearest CSL RX window,
+	 *    and before each use of IEEE_CONFIG_RX_SLOT setting parameters of the following (not
+	 *    the nearest one) CSL RX window, to allow the driver to calculate the proper
+	 *    CSL Phase to the nearest CSL window to inject in the CSL IEs for both transmitted
+	 *    data and ACK frames.
+	 * 3. Use @ref IEEE802154_CONFIG_CSL_PERIOD on each value change to update the current
+	 *    CSL period value which will be injected in the CSL IEs together with the CSL Phase
+	 *    based on @ref IEEE802154_CONFIG_CSL_RX_TIME.
```
Comment (kartben) [inline include/zephyr/net/ieee802154_radio.h:774]: very much a nit but does it need to be spelled "Phase" with a capital P?

File: include/zephyr/net/ieee802154.h [hunk 5]
```diff
@@ -171,24 +205,23 @@ struct ieee802154_security_ctx {
...
-	 * table 9-6 and ieee802154_security_level.
+	 * @brief Frame security level, possible values are defined in section
+	 * 9.4.2.2, table 9-6.
 	 *
-	 * @warning This is no longer in accordance with the current version of
-	 * the standard and needs to be extended in the future for full security
-	 * procedure compliance.
+	 * @warning Currently native L2 allows to configure one common security
+	 * level for all frame types, commands and information elements. This is
+	 * no longer in accordance with the current version of the standard and
```
Comment (kartben) [inline include/zephyr/net/ieee802154.h:217]: Here and a few other places: what is "current"??

**Review by kartben — DISMISSED** [review]

**Comment by kartben** [comment]
Besides the couple typos which would be nice to fix please don't assume any of my comments are blockers.

File: include/zephyr/net/ieee802154.h (hunk 2 above)
Comment (fgrandel) [reply include/zephyr/net/ieee802154.h]: Fixed everywhere (L2, net_mgmt, driver) I found it with `[Ww]e\b`.

File: include/zephyr/net/ieee802154.h (hunk 3 above)
Comment (fgrandel) [reply include/zephyr/net/ieee802154.h]: Fixed.

File: include/zephyr/net/ieee802154_radio.h (hunk 4 above)
Comment (fgrandel) [reply include/zephyr/net/ieee802154_radio.h]: Not at all. Just an oversight. Fixed.

File: include/zephyr/net/ieee802154.h (hunk 5 above)
Comment (fgrandel) [reply include/zephyr/net/ieee802154.h]: Good point. Specified the precise versions instead.

**Review by ghost — COMMENTED** [review]
Thanks for your detailed review @kartben :-) Very much appreciated! I pushed the rebase separately from the actual changes so that the compare button works for you.

@jukkar Could you re-ACK maybe, too? Had to push fixes.

**Review by jukkar — DISMISSED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jciupis — APPROVED** [review]

**Pull request merged**